distributed/worker.py

```diff
@@ -1028,9 +1013,52 @@ async def start(self):
         enable_gc_diagnosis()
         thread_state.on_event_loop_thread = True
 
-        await self.listen(
-            self._start_address, **self.security.get_listen_args("worker")
+        ports = parse_worker_ports(self._start_port)
```

Comment:
Slightly concerned with the amount of duplication between here and the Nanny. What are your thoughts?

Reply:
I spent some time trying to push this logic down into the underlying `Server.listen` call, but things got somewhat complicated so I decided to punt on it for now. I'll revisit this to see if we can do it in a simple way

Reply:
I'm completely fine with things as is if it's not straightforward to dedpulicate. I didn't look too closely.